specs/core/0_beacon-chain.md

```diff
@@ -2230,7 +2230,7 @@ def process_deposit(state: BeaconState, deposit: Deposit) -> None:
 
     # Verify the Merkle branch
     merkle_branch_is_valid = verify_merkle_branch(
-        leaf=hash(serialize(deposit.data)),  # 48 + 32 + 8 + 96 = 184 bytes serialization
+        leaf=hash_tree_root(deposit.data),
```

Comment:
Doesn't this mean that we also need to calculate the SSZ root of the deposit data in the Vyper contract?

Reply:
Yes, we need to calculate `hash_tree_root` in the contract. Assuming `def deposit(pubkey: bytes[48], withdrawal_credentials: bytes[32], amount: bytes[8], signature: bytes[96])` it's something like:

```python
pubkey_hash = hash(hash(pubkey[0:32]) + hash(pubkey[32:48] + zero_hash[0:16]))
signature_hash = hash(
    hash(hash(signature[0:32]) + hash(signature[32:64])) +
    hash(hash(signature[64:96]) + hash(zero_hash))
)
hash_tree_root = hash(
    hash(pubkey_hash + withdrawal_credentials) +
    hash(amount + zero_hash[0:24] + signature_hash)
)
```